doc/types/error_code.qbk

```diff
@@ -0,0 +1 @@
    as part of the C++ standard error handling utilities.  Except for the
    substitution of std:: for boost::system:: namespace, the libraries are
    identical.  This link points to the standard library documentation which
    may be used in addition to the boost version.
  ]]
  [[
    [@http://blog.think-async.com/2010/04/system-error-support-in-c0x-part-1.html
    Thinking Asyncronously in C++]
  ][
    Another essential reference on the design and usage of the `error_code`.
  ]]
]
[heading Associated Types]
[itemized_list
[
    [@http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Header-error_code `boost::system::error::ercc`]
    list of values which might be used as values by
    [@http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Class-error_code `boost::system::error_code`].
]
[
    [@../../include/nudb/error.hpp `nudb::error`]
    list of values which might be used as values NuDB library functions.
]
[
    [@http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Class-system_error `boost::system::error::system_error`]
]
]
Any NuDB function may return `error_code` values from either of the above sources.
So error_codes should be checked against both lists of error codes used.

[heading Notation]
[itemized_list
[`ec` an instance of a reference to the type `boost::system::error_code`.]
[<enum value> a value listed in either
`boost::system::errc` or
`nudb::error`.
]
]
[heading Valid Expressions]
[table
    [[Operation] [Return Type] [Description]]
    [
        [[@http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Class-error_code-observers `ec.message()`]]
        [`std::string`]
        [
            Return an explanation of the error code as a string
        ]
    ]
    [
        [[@http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Class-system_error `boost::system::error::system_error(ec)`]]
        [`boost::system::error::system_error`]
        [
            Construct a standard exception type.  The result can be the object
            of a C++ "throw" statement.
        ]
    ]
    [
        [`ec == `<enum value>]
        [bool]
        [
            Compare an error_code with an error enum.
        ]
    ]
    [
        [`ec != `<enum value>]
        [bool]
        [
            Compare an error_code with an error enum.
        ]
    ]
]

[heading Header]
[@http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Header-error_code `#include <boost/system/error_code.hpp>`]
[br]
[@../../include/nudb/error.hpp `#include <nudb/error.hpp>`]

[heading Example]
``
#include <boost/system/error_code.hpp>
#include <nudb/error.hpp>
// ...
boost::system::error_code ec;
nudb::store db;
db.open("db.dat", "db.key", "db.log", ec);
if(ec == nudb::error::no_key_file)
    std::cerr << ec.message() << std::endl;
if(ec == boost::system::errc::filename_too_long)
    std::cerr << ec.message() << std::endl;
// ...
``

[endsect]

```

Comment:
Weird `^M` happening here, and the file is missing a carriage return on the last line.